pixeltable/catalog/table.py

```diff
@@ -1550,3 +1552,131 @@ def __dir__(self) -> list[str]:
 
     def _ipython_key_completions_(self) -> list[str]:
         return list(self._schema.keys())
+
+    @classmethod
+    def diff_md(cls, old_md: Optional[dict[str, Any]], new_md: Optional[dict[str, Any]]) -> str:
+        """Return a string reporting the difference between two schema versions"""
+        assert new_md is not None
+        if old_md is None:
+            return 'Initial Version'
+        if old_md == new_md:
+            return ''
+        r = ''
+        added = {k: v['name'] for k, v in new_md.items() if k not in old_md}
+        if len(added) > 0:
+            r += 'Added: ' + ' '.join(added.values())
```

Comment:
Just FYI string concatenation in Python is slow. It won't matter in this case, but patterns of the form
```
r = ''
for obj in collection:
    r += 'a string for {obj}'
```
are a notorious source of performance bugs (that loop has quadratic running time). Using join is generally preferred:
```
c = []
for obj in collection:
    c.append('a string for {obj}')
r = ''.join(c)
```